model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -934,10 +935,19 @@ public Map<String, ClientScopeModel> getClientScopes(RealmModel realm, ClientMod
 
     @Override
     public Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String search, Integer first, Integer max) {
-        TypedQuery<String> query = em.createNamedQuery("getGroupIdsByNameContaining", String.class)
-                .setParameter("realm", realm.getId())
-                .setParameter("search", search);
+        return searchForGroupByNameStream(realm, search, false, first, max);
+    }
 
+    @Override
+    public Stream<GroupModel> searchForGroupByNameStream(RealmModel realm, String search, Boolean exact, Integer first, Integer max) {
+        TypedQuery<String> query;
+        if (exact.equals(Boolean.TRUE)) {
```

Comment:
NPE can be thrown here. Use the original suggestion.
```suggestion
        if (exact == Boolean.TRUE) {
```

Reply:
Before changing this I want to ask if it would be better to check for a null (perhaps even cast to false if null) rather than doing a direct object comparison. If I'm not mistaken, the direct object comparison means any case where `exact` is not precisely the object `Boolean.TRUE` (as in they have the exact same object ID) will fail a comparison regardless of truthiness

The direct object comparison would mean enforcing that every call to this method passes in either `Boolean.TRUE` or `Boolean.FALSE`, correct?

Reply:
Yeah unfortunately not all boxed booleans are the same instance. A failsafe approach I think is `Boolean.TRUE.equals(exact)`

Reply:
Updated usage to use the above suggestion, avoiding the object comparison as well as the NPE, advise if this is incorrect